react/ProductHighlights.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { FC, useContext, ReactNode, useMemo } from 'react'
+import { useProduct } from 'vtex.product-context'
+
+import { getSeller } from './modules/seller'
+
+type HighlightType = 'collection' | 'promotion' | 'teaser'
+
+interface ProductHighlightsProps {
+  filter?: Filter
+  type?: HighlightType
+  children: ReactNode
```

Comment:
⛏️ 
```suggestion
```

Reply:
Or he could create a `function ProductHighlights(props: ProductHighlightsProps)` 😏

Reply:
@victorhmp @Klynger 
I need this because children is required

Reply:
If `children` is required so this is ambiguous I think. Because you're declaring here and with `React.FC`, another point is that `React.ReactNode` could be `null` and it seems that you don't wanna that, here is the type of `React.ReactNode`

```typescript
type ReactNode = ReactChild | ReactFragment | ReactPortal | boolean | null | undefined;
```

Reply:
Maybe you could define it as `Exclude<ReactNode, undefined | null>`? 
But I'm not sure you can always expect `children` to be there 🤔

Reply:
Why not @victorhmp ?

Reply:
I don't have a good reason, just seems strange to me 😅.
Anyway, I guess the type definition above would make your intention clearer 😄

Reply:
I'll just remove this type definition, it's OK if the user doesn't pass a children.

(btw, there is [NonNullable<Type>](https://www.typescriptlang.org/docs/handbook/utility-types.html#nonnullabletype) that I could use here)